packages/components/src/components/es-tabs/es-tabs.tsx

```diff
@@ -37,79 +41,96 @@ export class EsTabs {
 
     @State() activeDragOver?: string;
 
     /** Triggered when the active tab is changed. `detail` is the newly active tab. */
     @Event() tabChange!: EventEmitter<string>;
 
     private searchParam?: SearchParamTracker;
+    private observer = new ResizeObserver(() => {
+        this.positionIndicator(this.active);
+    });
 
     componentWillLoad() {
-        this.searchParam = this.activeParam
-            ? searchParam(this.activeParam)
-            : undefined;
-
-        if (this.active && this.searchParam && !this.searchParam.value) {
-            this.searchParam.set(this.active);
-        }
-
-        this.active = this.searchParam?.value ?? this.active ?? this.tabs[0].id;
+        this.setUpActiveParam(this.activeParam);
     }
 
     disconnectedCallback() {
+        this.observer.disconnect();
         this.searchParam?.delete();
-        cancelAnimationFrame(this.frame1);
-        cancelAnimationFrame(this.frame2);
+        cancelAnimationFrame(this.indicatorFrame);
         this.dragTimeouts.forEach((t) => clearTimeout(t));
     }
 
-    @Watch('active') updateActive(active?: string) {
-        // workaround for https://github.com/ionic-team/stencil/issues/2982
-        if (active) {
-            const slot = this.host.shadowRoot?.querySelector('slot');
+    @Watch('activeParam')
+    setUpActiveParam(param: string | false) {
+        if (!param) {
+            this.searchParam?.delete();
+            this.searchParam = undefined;
+        } else {
+            this.searchParam = searchParam(param);
 
-            if (slot) {
-                slot.name = active;
+            if (this.active && this.searchParam && !this.searchParam.value) {
+                this.searchParam.set(this.active);
             }
         }
 
+        this.active = this.searchParam?.value ?? this.active ?? this.tabs[0].id;
+    }
+
+    @Watch('active')
+    updateActive(active?: string) {
         this.searchParam?.set(active);
         this.tabChange.emit(active);
     }
 
-    renderTab = ({ title, id, badge }: Tab) => (
-        <button
-            data-skip-focus-delegation
-            type={'button'}
-            role={'tab'}
-            aria-controls={id}
-            aria-selected={this.active === id}
-            id={`tab-${id}`}
-            class={{
-                tab: true,
-                active: this.active === id,
-                drag_over: this.activeDragOver === id,
-            }}
-            part={this.active === id ? 'tab active' : 'tab'}
-            onClick={this.setActive(id)}
-            ref={this.captureTab(id)}
-            onDragEnter={this.dragEnterTab(id)}
-            onDragLeave={this.dragLeaveTab(id)}
-            onDrop={this.dropTab(id)}
-            onDragOver={this.dragOverTab}
-        >
-            <es-badge count={badge?.() ? 1 : 0} variant={'dot'}>
-                {title}
-            </es-badge>
-        </button>
+    renderTab = ({ title, id, badge }: Tab, i: number) => (
+        <>
```

Comment:
Unlike React, StencilJS does not support an equivalent to React.Fragment, which would allow us to group elements without adding extra DOM nodes. Consequently, we are unable to assign a 'key' directly on a fragment-like element here. This is generally fine under most circumstances, but could potentially lead to issues with element reordering or state persistence in dynamic lists if the components are significantly modified in the future. If such problems arise, consider restructuring or adding keys at a higher level component.